src/ripple/rpc/handlers/AccountTx.cpp

```diff
@@ -69,8 +69,10 @@ Json::Value doAccountTx (RPC::Context& context)
         std::int64_t iLedgerMax  = params.isMember ("ledger_index_max")
                 ? params["ledger_index_max"].asInt () : -1;
 
-        uLedgerMin  = iLedgerMin == -1 ? uValidatedMin : iLedgerMin;
-        uLedgerMax  = iLedgerMax == -1 ? uValidatedMax : iLedgerMax;
+        uLedgerMin  = iLedgerMin == -1 ? uValidatedMin :
+            ((iLedgerMin >= uValidatedMin) ? iLedgerMin : uValidatedMin);
+        uLedgerMax  = iLedgerMax == -1 ? uValidatedMax :
+            ((iLedgerMax <= uValidatedMax) ? iLedgerMax : uValidatedMax);
```

Comment:
Interesting.  C++ Standard section 4.7 says conversion from `int64_t` to `uint32_t` is legal and does exactly what you want here.  Code looks good to me.